lib/init/features/firestore.js

```diff
@@ -27,11 +33,60 @@ var _initRules = function(setup, config) {
       type: "input",
       name: "rules",
       message: "What file should be used for Firestore Rules?",
-      default: "firestore.rules",
+      default: DEFAULT_RULES_FILE,
     },
-  ]).then(function() {
-    return config.writeProjectFile(setup.config.firestore.rules, RULES_TEMPLATE);
-  });
+  ])
+    .then(function() {
+      var filename = setup.config.firestore.rules;
+
+      if (fsutils.fileExistsSync(filename)) {
+        var msg =
+          "File " +
+          chalk.bold(filename) +
+          " already exists." +
+          " Do you want to overwrite it with the Firestore Rules from the Firebase Console?";
+        return prompt.once({
+          type: "confirm",
+          message: msg,
+          default: false,
+        });
+      }
+
+      return Promise.resolve(true);
+    })
+    .then(function(overwrite) {
+      if (!overwrite) {
+        return Promise.resolve();
+      }
+
+      return _getRulesFromConsole(setup.projectId).then(function(contents) {
+        return config.writeProjectFile(setup.config.firestore.rules, contents);
+      });
+    });
+};
+
+var _getRulesFromConsole = function(projectId) {
+  return gcp.rules
+    .getLatestRulesetName(projectId)
+    .then(function(name) {
+      if (!name) {
+        logger.debug("No rulesets found, using default.");
+        return [{ name: DEFAULT_RULES_FILE, content: RULES_TEMPLATE }];
+      }
+
+      logger.debug("Found ruleset: " + name);
+      return gcp.rules.getRulesetContent(name);
+    })
+    .then(function(rules) {
+      for (var i = 0; i < rules.length; i++) {
+        var entry = rules[i];
+        if (entry.name === DEFAULT_RULES_FILE) {
```

Comment:
Could you add a comment to explain what's going on?

Reply:
Hmm ok actually you are right.  It doesn't seem to matter what the name is (I deployed `custom.rules`). 

So I will make a chance.